### PR TITLE
Improved the error message when HTTP proxies reject requests from Pulp.

### LINKED ISSUE #2654: Improve proxy connection failure error message
Raise a more informative error message when the proxy rejects requests from Pulp

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2047485

### PR DESCRIPTION
**Pull request by dralley**
closes #2654

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

**Comment by dralley** [comment]
Tested by Joniel w/ a manual patch - see private BZ comments.

File: pulpcore/download/http.py [hunk 1]
```diff
@@ -256,6 +256,16 @@ async def download_wrapper():
                     return await self._run(extra_data=extra_data)
                 except asyncio.TimeoutError:
                     raise TimeoutException(self.url)
+                except aiohttp.ClientHttpProxyError as e:
+                    log.error(
+                        "Proxy {!r} rejected connection request during a request to {!r}, status={}, message={!r}".format(
+                            e.request_info.real_url,
```
Comment (dralley) [inline pulpcore/download/http.py:262]: Lol.

**Comment by dralley** [comment]
Note to self, we probably shouldn't fall back to attempting to use the url as a mirrorlist unless the response code is 403 or 404.

**Review by ggainey — APPROVED** [review]

**Review by dkliban — APPROVED** [review]

**Comment by dralley** [comment]
Backporting as far as 3.16, but no further.

**Pull request merged**